api/src/main/resources/dao/datastorage-tag-dao.xml

```diff
@@ -176,5 +176,28 @@
                 ]]>
             </value>
         </constructor-arg>
+        <constructor-arg>
+            <value>
+                <!--searchDatastorageObjectsByTagQuery-->
+                <![CDATA[
+                    SELECT *
+                    FROM pipeline.datastorage_tag t
+                    WHERE t.tag_key = :TAG_KEY
+                    AND (:TAG_VALUE::text IS NULL OR t.tag_value = :TAG_VALUE)
+                ]]>
+            </value>
+        </constructor-arg>
+        <constructor-arg>
+            <value>
+                <!--searchDatastorageObjectsByTagInSpecificStoragesQuery-->
+                <![CDATA[
+                    SELECT *
+                    FROM pipeline.datastorage_tag t
+                    WHERE t.tag_key = :TAG_KEY
+                    AND (:TAG_VALUE::text IS NULL OR t.tag_value = :TAG_VALUE)
+                    AND t.datastorage_root_id IN (:DATASTORAGE_ROOT_ID)
+                ]]>
+            </value>
+        </constructor-arg>
```

Comment:
Is it possible to use just a single sql query for both cases? By checking *DATASTORAGE_ROOT_ID* nullability similarly to *TAG_VALUE*.